### PR TITLE
Add Shank IDL to Program ELF

### PR DESCRIPTION
**Pull request by ngundotra**
Add the compressed IDL to the program's binary under a section header called `.solana.idl`. 

Adding `include_idl!()` increases program size by `4K` bytes.
* Before including compressed IDL, `asset` program size: `244K`
* After including .solana.idl, `asset` program size: `248K`

### CODE REVIEW

**Comment by febo** [comment]
@ngundotra This is super cool!

I think we need to instal `node + pnpm` on the `lint_rust` step for CI to work. Probably all we need is this block:
```
- name: Install Node.js
  uses: nifty-oss/actions/install-node-with-pnpm@v1
  with:
     version: ${{ env.NODE_VERSION }}
     cache: ${{ env.CACHE }}
     dependencies: true
```
on the `lint_rust` step on `main.yml` after installing Rust.

**Comment by ngundotra** [comment]
@febo I could use some help fixing the rust client tests :) when you have a moment

**Comment by ngundotra** [comment]
@febo I could use some help fixing the rust client tests :) when you have a moment

**Comment by ngundotra** [comment]
@febo I could use some help fixing the rust client tests :) when you have a moment

**Comment by febo** [comment]
> @febo I could use some help fixing the rust client tests :) when you have a moment

Having a look now.

File: programs/asset/program/Cargo.toml [hunk 1]
```diff
@@ -6,6 +6,8 @@ edition = "2021"
 readme = "./README.md"
 license-file = "Apache-2.0"
 publish = false
+build = "build.rs"
+build-sbf = "build.rs"
```
Comment (febo) [inline programs/asset/program/Cargo.toml:10]: It seems that the `build-sbf` can be removed. There is a warning saying that it is not used.

**Comment by febo** [comment]
@ngundotra We need to add the `node + pnpm` install action to both build and test workflows:
```
- name: Install Node.js
  uses: nifty-oss/actions/install-node-with-pnpm@v1
  with:
     version: ${{ env.NODE_VERSION }}
     cache: ${{ env.CACHE }}
     dependencies: true
```
Similarly to what was done for lint.

* `build-programs.yml`: https://github.com/nifty-oss/asset/blob/main/.github/workflows/build-programs.yml#L42-L48
* `test-programs.yml`: https://github.com/nifty-oss/asset/blob/main/.github/workflows/test-programs.yml#L31-L37

**Review by febo — APPROVED** [review]

**Pull request merged**